src/burn/engine/variable.cpp

```diff
@@ -1961,6 +1964,40 @@ static HRESULT InitializeVariableRebootPending(
         }
     }
 
+    hr = RegKeyReadNumber(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Microsoft\\ServerManager", L"CurrentRebootAttempts", TRUE, &dwValue);
+    fRebootPending |= SUCCEEDED(hr) && 0 < dwValue;
```

Comment:
How about a skipping all the tests as soon as one comes back true? No point doing all the checks if first one passes.